src/xunit.performance.api/XunitBenchmark.cs

```diff
@@ -14,23 +14,25 @@ static XunitBenchmark()
             // TODO: Read this from file.
             //   1. This reads as: flags read from config file based on running platform!
             //   2. Format could be something like this: `{ "Name" : "BranchMispredictions", "interval" : "100000" }`
-            RequiredProviders = new ProviderInfo[]
+            PerformanceMonitorCounters = new List<IPerformanceMonitorCounter>
+            {
+                new BranchMispredictionsPerformanceMonitorCounter(),
+                new CacheMissesPerformanceMonitorCounter(),
+            };
+
+            PerformanceMetricInfos = new List<BasePerformanceMetric>
```

Comment:
IPerformanceMonitorCounter is an interface that defines the configuration of the pmc to be used, not the internal of the pmc (at least for now). The idea is to have a user interface decoupled from the implementation of the pmc (the pmc id). Currently, this is hardcoded in the function, but you can imagine that this data is given by the API user instead (in the near future).